staging/src/k8s.io/component-base/metrics/testutil/metrics.go

```diff
@@ -360,3 +360,20 @@ func GetHistogramMetricValue(m metrics.ObserverMetric) (float64, error) {
 	}
 	return metricProto.Histogram.GetSampleSum(), nil
 }
+
+// LabelsMatch returns true if metric has all expected labels otherwise false
+func LabelsMatch(metric *dto.Metric, labelFilter map[string]string) bool {
+	for labelName, labelValue := range labelFilter {
+		labelFound := false
+		for _, labelPair := range metric.Label {
+			if labelName == *labelPair.Name && labelValue == *labelPair.Value {
+				labelFound = true
+				break
+			}
+		}
```

Comment:
nit: we can bring the nested for-loop to the outer scope to avoid the repeated work (practically the number of labels are going to be bounded in tests and hopefully in our metrics).

```
metricLabels := map[string]string{}
for _, labelPair := range metric.Label {
	metricLabels[labelPair.GetName()] = labelPair.GetValue()
}
// length comparison then match key to values in the maps
```

Reply:
/cc @logicalhan 
updated